### PR TITLE
OAS-3737 | add support for setting disk autosize settings

### PR DESCRIPTION
**Pull request by forgems**

### CODE REVIEW

File: cmd/data/create_deployment.go [hunk 1]
```diff
@@ -18,6 +18,7 @@
 // Copyright holder is ArangoDB GmbH, Cologne, Germany
 //
 // Author Ewout Prangsma
+// Author Marcin Swiderski
```
Comment (ewoutp) [inline cmd/data/create_deployment.go:21]: Drop both

File: cmd/data/create_deployment.go [hunk 2]
```diff
@@ -164,6 +177,21 @@ func init() {
 						EmailAddresses: cargs.notificationEmailAddresses,
 					}
 				}
+				if !c.Flags().Changed("max-node-disk-size") {
+					cargs.maxNodeDiskSize = cargs.nodeDiskSize * 2
```
Comment (ewoutp) [inline cmd/data/create_deployment.go:181]: Do we really need this?
I would assume that if it is not set at all, the backend takes care of the default

File: cmd/data/update_deployment.go [hunk 3]
```diff
@@ -18,6 +18,7 @@
 // Copyright holder is ArangoDB GmbH, Cologne, Germany
 //
 // Author Ewout Prangsma
+// Author Marcin Swiderski
```
Comment (ewoutp) [inline cmd/data/update_deployment.go:21]: Drop both

File: cmd/data/update_deployment.go [hunk 4]
```diff
@@ -158,6 +167,10 @@ func init() {
 					ensureModel().NodeDiskSize = cargs.nodeDiskSize
 					hasChanges = true
 				}
+				if f.Changed("max-node-disk-size") {
+					ensureDiskAutosizeSettings().MaximumNodeDiskSize = cargs.maxNodeDiskSize
```
Comment (ewoutp) [inline cmd/data/update_deployment.go:171]: Don't we need a check for nodeSize.MaxDiskSize here?

File: pkg/format/deployment.go [hunk 5]
```diff
@@ -18,6 +18,7 @@
 // Copyright holder is ArangoDB GmbH, Cologne, Germany
 //
 // Author Ewout Prangsma
+// Author Marcin Swiderski
```
Comment (ewoutp) [inline pkg/format/deployment.go:21]: Drop both

**Review by ewoutp — CHANGES_REQUESTED** [review]

File: cmd/data/create_deployment.go (hunk 1 above)
Comment (forgems) [reply cmd/data/create_deployment.go]: Dropped.

File: cmd/data/create_deployment.go (hunk 2 above)
Comment (forgems) [reply cmd/data/create_deployment.go]: No need for that. I forgot that  we have this logic in backend

File: cmd/data/update_deployment.go (hunk 3 above)
Comment (forgems) [reply cmd/data/update_deployment.go]: dropped

File: cmd/data/update_deployment.go (hunk 4 above)
Comment (forgems) [reply cmd/data/update_deployment.go:168]: It's implemented on backend

File: pkg/format/deployment.go (hunk 5 above)
Comment (forgems) [reply pkg/format/deployment.go]: done

**Review by ewoutp — APPROVED** [review]
LGTM

**Pull request merged**